### PR TITLE
More informative error message on boolean images for regionprops 

### PR DESCRIPTION
**Pull request by bioimage-analysis**
## Description

First time contribution fixing issue #3668 

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by bioimage-analysis** [comment]
Thank you Emmanuelle for your comment, sorry, I was not paying attention, I was testing how to use Github to contribute to scikit image on something relatively simple...

File: skimage/measure/_regionprops.py [hunk 1]
```diff
@@ -833,7 +833,11 @@ def regionprops(label_image, intensity_image=None, cache=True,
         raise TypeError('Only 2-D and 3-D images supported.')
 
     if not np.issubdtype(label_image.dtype, np.integer):
-        raise TypeError('Label image must be of integer type.')
+        raise TypeError('Non-integer image types are ambiguous:'
+                        'use ndimage.label to label the connected'
```
Comment (soupault) [inline skimage/measure/_regionprops.py]:
```suggestion
                        'use skimage.measure.label to label the connected '
```

File: skimage/measure/_regionprops.py [hunk 2]
```diff
@@ -833,7 +833,11 @@ def regionprops(label_image, intensity_image=None, cache=True,
         raise TypeError('Only 2-D and 3-D images supported.')
 
     if not np.issubdtype(label_image.dtype, np.integer):
-        raise TypeError('Label image must be of integer type.')
+        raise TypeError('Non-integer image types are ambiguous:'
```
Comment (soupault) [inline skimage/measure/_regionprops.py]:
```suggestion
        raise TypeError('Non-integer image types are ambiguous: '
```

File: skimage/measure/_regionprops.py [hunk 3]
```diff
@@ -833,7 +833,11 @@ def regionprops(label_image, intensity_image=None, cache=True,
...
+                        'use ndimage.label to label the connected'
+                        'components of the image,'
```
Comment (soupault) [inline skimage/measure/_regionprops.py]:
```suggestion
                        'components of the label_image, '
```

File: skimage/measure/_regionprops.py [hunk 4]
```diff
@@ -833,7 +833,11 @@ def regionprops(label_image, intensity_image=None, cache=True,
...
+                        'or label_image.astype(np.uint8) to interpret'
```
Comment (soupault) [inline skimage/measure/_regionprops.py]:
```suggestion
                        'or label_image.astype(np.uint8) to interpret '
```

File: skimage/measure/_regionprops.py [hunk 5]
```diff
@@ -833,7 +833,11 @@ def regionprops(label_image, intensity_image=None, cache=True,
...
+                        'the True values as a single label')
```
Comment (soupault) [inline skimage/measure/_regionprops.py]:
```suggestion
                        'the True values as a single label.')
```

**Comment by emmanuelle** [comment]
@bioimage-analysis since your PR was almost finished I pushed a commit to your branch and hope to have it merged soon. You don't have anything to do unless you want to :-) (for this PR or maybe future contributions ;-). Thanks !

**Comment by hmaarrfk** [comment]
Are we OK with having this fail as a hard error? Should we not deprecate this feature using the regular cycle:

1. Warn for 2 releases.
2. Then in 0.19 make it fail as an error?

**Comment by emmanuelle** [comment]
@hmaarrfk throwing an error is already the current behaviour so I don't think we need to go through a deprecation cycle, we just give a more helpful error message here in the case of boolean images.

**Comment by emmanuelle** [comment]
Thanks for the approval @hmaarrfk ; since I was the one to adapt this PR I don't count as 2nd +1, so we'll need another one :-).

**Comment by jni** [comment]
@bioimage-analysis @hmaarrfk @emmanuelle thank you all for your work here! This is a significant usability improvement. =)

**Pull request merged**